pkg/gardenlet/controller/controllerinstallation/controllerinstallation_care_control_test.go

```diff
@@ -0,0 +1,233 @@
+				RequeueAfter: syncPeriodDuration,
+			}))
+
+			Expect(gardenClient.Get(ctx, client.ObjectKeyFromObject(controllerInstallation), controllerInstallation)).To(Succeed())
+			Expect(controllerInstallation.Status.Conditions).To(consistOfConditionsInUnknownStatus("SeedReadError", "Failed to get ManagedResource"))
+		})
+
+		DescribeTable("should set correct conditions when managed resource exists", func(managedResource *resourcesv1alpha1.ManagedResource, matchExpectedConditions gomegatypes.GomegaMatcher) {
+			Expect(seedClient.Create(ctx, managedResource)).To(Succeed())
+
+			result, err := reconciler.Reconcile(ctx, request)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(Equal(reconcile.Result{
+				RequeueAfter: syncPeriodDuration,
+			}))
+
+			Expect(gardenClient.Get(ctx, client.ObjectKeyFromObject(controllerInstallation), controllerInstallation)).To(Succeed())
+			Expect(controllerInstallation.Status.Conditions).To(matchExpectedConditions)
+		},
+			Entry("managed resource conditions are not set",
+				managedResource(nil),
+				ConsistOf(
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionFalse, "InstallationPending"),
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionFalse, "ControllerNotHealthy"),
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionTrue, "ControllerNotRolledOut"),
+				),
+			),
+			Entry("managed resource is not healthy",
+				notHealthyManagedResource(),
+				ConsistOf(
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionFalse, "InstallationPending"),
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionFalse, "ControllerNotHealthy"),
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionTrue, "ControllerNotRolledOut"),
+				),
+			),
+			Entry("managed resource is healthy",
+				healthyManagedResource(),
+				ConsistOf(
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionTrue, "InstallationSuccessful"),
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionTrue, "ControllerHealthy"),
+					conditionWithStatusAndReason(gardencorev1beta1.ConditionFalse, "ControllerRolledOut"),
+				),
+			),
+		)
+	})
+})
+
+func consistOfConditionsInUnknownStatus(reason, message string) gomegatypes.GomegaMatcher {
+	return ConsistOf(
+		conditionWithStatusReasonAndMesssage(gardencorev1beta1.ConditionUnknown, reason, message),
+		conditionWithStatusReasonAndMesssage(gardencorev1beta1.ConditionUnknown, reason, message),
+		conditionWithStatusReasonAndMesssage(gardencorev1beta1.ConditionUnknown, reason, message),
+	)
+}
+
+func conditionWithStatusAndReason(status gardencorev1beta1.ConditionStatus, reason string) gomegatypes.GomegaMatcher {
+	return conditionWithStatusReasonAndMesssage(status, reason, "")
+}
+
+func conditionWithStatusReasonAndMesssage(status gardencorev1beta1.ConditionStatus, reason, message string) gomegatypes.GomegaMatcher {
+	return MatchFields(IgnoreExtras, Fields{
+		"Status":  Equal(status),
```

Comment:
Nit: we could add the type field as well to make sure we are asserting the right condition type.